loom/dispatchers/LAWProtocolDispatcher.py

```diff
@@ -83,6 +85,26 @@ class LAWBadArgumentsError(LAWError):
     """
     pass
 
+class LAWNotLoggedInError(LAWError):
+    """
+    Raised when an action is requested without having logged in.
+    """
+    pass
+
+############################################################
+#
+# LAWProtocolDispatcher Decorators
+#
+############################################################
+
+
+@decorator
+def requires_login(func, *args, **kwargs):
+    self = args[0]
```

Comment:
I don't like this, but I think we'll have to go with it for now. However, we should investigate alternative methodologies in the future.